lib/systems/platforms.nix

```diff
@@ -385,16 +403,15 @@ rec {
       '';
       target = "Image";
     };
-    gcc = {
-      arch = "armv8-a";
+    cpuModel = {
+      gnu = "armv8-a";
+      llvm = "armv8-a";
+      zig = "armv8-a";
     };
   };
 
   apple-m1 = {
-    gcc = {
-      arch = "armv8.3-a+crypto+sha2+aes+crc+fp16+lse+simd+ras+rdm+rcpc";
-      cpu = "apple-a13";
-    };
+    cpuModel = "apple_m1";
```

Comment:
This should fix https://github.com/NixOS/nixpkgs/issues/166401? The reason for the lengthy arch string was due to LLVM 11, which isn’t even in nixpkgs anymore.

Reply:
Likely, needs testing though.